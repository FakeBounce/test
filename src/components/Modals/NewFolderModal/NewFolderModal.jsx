import React, { Component } from "react";
import { Button } from "reactstrap";
import { observer } from "mobx-react";
import { observable } from "mobx";
import { translate } from "common/methods/translations";

import Input from "elements/Input/Input";
import Modal from "elements/Modal/Modal";
import styles from "../FormModal/FormModal.module.scss";

@observer
class NewFolderModal extends Component {
  @observable error = false;

  constructor(props) {
    super(props);
    this.state = {
      fileName: props.currentName || ""
    };
    this.handleFileNameChange = this.handleFileNameChange.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.currentName) {
      this.setState(state => ({
        ...state,
        fileName: nextProps.currentName
      }));
    }
  }

  saveButtonClick() {
    const { fileName } = this.state;
    if (fileName !== "") {
      this.error = false;
      const { handleYesButtonClick } = this.props;
      handleYesButtonClick(fileName);
      this.setState(state => ({
        ...state,
        fileName: ""
      }));
    } else {
      this.error = true;
    }
  }

  cancelButtonClick() {
    const { handleNoButtonClick } = this.props;
    handleNoButtonClick();
    this.setState(state => ({
      ...state,
      fileName: ""
    }));
  }

  saveCancelButtons() {
    const { isRenaming } = this.props;
    return (
      <div>
        <Button color="primary" onClick={this.saveButtonClick.bind(this)}>
          {isRenaming ? translate("globals.save") : translate("globals.create")}
        </Button>
        <Button color="secondary" onClick={this.cancelButtonClick.bind(this)}>
          {translate("globals.cancel")}
        </Button>
      </div>
    );
  }

  handleFileNameChange(fileName) {
    this.setState(state => ({
      ...state,
      fileName: fileName
    }));
  }

  render() {
    const { fileName } = this.state;
    return (
      <Modal
        ModalFooterContent={this.saveCancelButtons.bind(this)}
        {...this.props}
        className={"NewFolderModal"}
      >
        <Input
          inputValue={this.handleFileNameChange}
          placeholder={translate("globals.fileManager.folderName")}
          initialValue={fileName}
        />
        <p className={styles.ErrorBox} hidden={!this.error}>
          {translate("globals.required")}
        </p>
      </Modal>
    );
  }
}

export default NewFolderModal;
