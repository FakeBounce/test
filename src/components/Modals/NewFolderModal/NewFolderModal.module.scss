.ErrorBox {
  font-size: 12px;
  color: #FF0000;
  margin-left: 15px;
}
