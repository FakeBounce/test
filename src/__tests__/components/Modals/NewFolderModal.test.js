import React from "react";
import { shallow, configure } from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";
import NewFolderModal from "components/Modals/NewFolderModal/NewFolderModal";

configure({ adapter: new Adapter() });

const handleYes = jest.fn();
const handleNo = jest.fn();

const defaultTestingProps = {
  isOpen: false,
  title: "Title",
  handleYesButtonClick: handleYes,
  handleNoButtonClick: handleNo,
  toggle: () => jest.fn(),
  isRenaming: false
};

describe("Testing NewFolderModal component", () => {
  it("renders as expected without parameters", () => {
    const wrapper = shallow(<NewFolderModal />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders as expected with parameters (as closed)", () => {
    const wrapper = shallow(<NewFolderModal {...defaultTestingProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders as expected with parameters (as open)", () => {
    const wrapper = shallow(<NewFolderModal {...defaultTestingProps} isOpen />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders as expected with parameters (as isRenaming)", () => {
    const wrapper = shallow(
      <NewFolderModal {...defaultTestingProps} isRenaming />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("behave correctly when calling props", () => {
    const wrapper = shallow(
      <NewFolderModal {...defaultTestingProps} currentName={"File"} />
    );

    wrapper.instance().saveButtonClick();
    expect(handleYes.mock.calls.length).toBe(1);
    // expect(handleYes.mock.calls.length).toBeCalledWith(false);

    wrapper.instance().cancelButtonClick();
    expect(handleNo.mock.calls.length).toBe(1);
    // expect(handleYes.mock.calls.length).toBeCalledWith(false);
  });

  it("behave correctly when props are invalid", () => {
    const wrapper = shallow(
      <NewFolderModal
        {...defaultTestingProps}
        currentName={"File"}
        handleYesButtonClick={true}
        handleNoButtonClick={true}
      />
    );
    try {
      wrapper.instance().saveButtonClick();
      // Fail test if above expression doesn't throw anything.
      expect(true).toBe(false);
    } catch (e) {
      expect(e).toBeInstanceOf(TypeError);
      expect(e.message).toBe("handleYesButtonClick is not a function");
    }
    try {
      wrapper.instance().cancelButtonClick();
      // Fail test if above expression doesn't throw anything.
      expect(true).toBe(false);
    } catch (e) {
      expect(e).toBeInstanceOf(TypeError);
      expect(e.message).toBe("handleNoButtonClick is not a function");
    }
  });

  // @TODO Try to access to wrapper to fullfill test
  it("behave correctly when updating state", () => {
    const wrapper = shallow(<NewFolderModal {...defaultTestingProps} />);

    wrapper.instance().handleFileNameChange("New name");
    expect(wrapper.state().fileName).toEqual("New name");
  });
});
