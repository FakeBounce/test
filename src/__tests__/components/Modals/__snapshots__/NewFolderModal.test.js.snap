// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NewFolderModal component renders as expected with parameters (as closed) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewFolderModal
    handleNoButtonClick={[Function]}
    handleYesButtonClick={[Function]}
    isOpen={false}
    isRenaming={false}
    title="Title"
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "children": Array [
        <InputElement
          initialValue=""
          inputValue={[Function]}
          placeholder="Nom du dossier"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
      ],
      "className": "NewFolderModal",
      "handleNoButtonClick": [Function],
      "handleYesButtonClick": [Function],
      "isOpen": false,
      "isRenaming": false,
      "title": "Title",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "initialValue": "",
          "inputValue": [Function],
          "placeholder": "Nom du dossier",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "children": Array [
          <InputElement
            initialValue=""
            inputValue={[Function]}
            placeholder="Nom du dossier"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
        ],
        "className": "NewFolderModal",
        "handleNoButtonClick": [Function],
        "handleYesButtonClick": [Function],
        "isOpen": false,
        "isRenaming": false,
        "title": "Title",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "initialValue": "",
            "inputValue": [Function],
            "placeholder": "Nom du dossier",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing NewFolderModal component renders as expected with parameters (as isRenaming) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewFolderModal
    handleNoButtonClick={[Function]}
    handleYesButtonClick={[Function]}
    isOpen={false}
    isRenaming={true}
    title="Title"
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "children": Array [
        <InputElement
          initialValue=""
          inputValue={[Function]}
          placeholder="Nom du dossier"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
      ],
      "className": "NewFolderModal",
      "handleNoButtonClick": [Function],
      "handleYesButtonClick": [Function],
      "isOpen": false,
      "isRenaming": true,
      "title": "Title",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "initialValue": "",
          "inputValue": [Function],
          "placeholder": "Nom du dossier",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "children": Array [
          <InputElement
            initialValue=""
            inputValue={[Function]}
            placeholder="Nom du dossier"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
        ],
        "className": "NewFolderModal",
        "handleNoButtonClick": [Function],
        "handleYesButtonClick": [Function],
        "isOpen": false,
        "isRenaming": true,
        "title": "Title",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "initialValue": "",
            "inputValue": [Function],
            "placeholder": "Nom du dossier",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing NewFolderModal component renders as expected with parameters (as open) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewFolderModal
    handleNoButtonClick={[Function]}
    handleYesButtonClick={[Function]}
    isOpen={true}
    isRenaming={false}
    title="Title"
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "children": Array [
        <InputElement
          initialValue=""
          inputValue={[Function]}
          placeholder="Nom du dossier"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
      ],
      "className": "NewFolderModal",
      "handleNoButtonClick": [Function],
      "handleYesButtonClick": [Function],
      "isOpen": true,
      "isRenaming": false,
      "title": "Title",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "initialValue": "",
          "inputValue": [Function],
          "placeholder": "Nom du dossier",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "children": Array [
          <InputElement
            initialValue=""
            inputValue={[Function]}
            placeholder="Nom du dossier"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
        ],
        "className": "NewFolderModal",
        "handleNoButtonClick": [Function],
        "handleYesButtonClick": [Function],
        "isOpen": true,
        "isRenaming": false,
        "title": "Title",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "initialValue": "",
            "inputValue": [Function],
            "placeholder": "Nom du dossier",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing NewFolderModal component renders as expected without parameters 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewFolderModal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "children": Array [
        <InputElement
          initialValue=""
          inputValue={[Function]}
          placeholder="Nom du dossier"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
      ],
      "className": "NewFolderModal",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "initialValue": "",
          "inputValue": [Function],
          "placeholder": "Nom du dossier",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "children": Array [
          <InputElement
            initialValue=""
            inputValue={[Function]}
            placeholder="Nom du dossier"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
        ],
        "className": "NewFolderModal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "initialValue": "",
            "inputValue": [Function],
            "placeholder": "Nom du dossier",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;
